### ISSUE TITLE
Tabstops on the fly

### ISSUE THREAD

**Issue by alivefromupstairs** [Microsoft/vscode#72926]
Create tabstops like you would find in snippets by adding cursors around. Would be very helpful to navigate and type data with a pre-defined path.

![image](https://user-images.githubusercontent.com/46059092/56819803-6c8fbb80-6842-11e9-80c6-92d1743d1382.png)

The last one should be the equivalent of `$0` in snippets.

**Comment by rbenson** [comment]
@alexandrudima - this was asked in another forum, and https://marketplace.visualstudio.com/items?itemName=danseethaler.tab-through-selections met the need. Recommend closing, unless you want to look into adding this natively.

**Comment by alivefromupstairs** [comment]
@rbenson @alexandrudima yes, the extension doesn't have highlighting nor can you go back and forth and edit something incomplete; it messes up the order.

From the extension's repo:
>Content edits during tabbing through selections can offset the underlying content making the selections invalid. We need something similar to markers in atom to keep track of the selection positions. I haven't found anything like this yet with the VSCode api.

**Comment by hediet** [comment]
> Content edits during tabbing through selections can offset the underlying content making the selections invalid. We need something similar to markers in atom to keep track of the selection positions. I haven't found anything like this yet with the VSCode api.

This can be implemented by the extension.
[Here](https://github.com/leanprover/vscode-lean/blob/master/src/abbreviation/rewriter/TrackedAbbreviation.ts) is an example how to do it.
